addons/website_blog/views/website_blog_components.xml

```diff
@@ -21,9 +22,61 @@
+                <a t-attf-class="btn btn-light #{not wblog_nav_offcanvas and 'dropdown-toggle d-flex align-items-center justify-content-between'}"
+                   role="button"
+                   t-attf-title="{{blog.name if blog else wblog_all_title_string}}"
+                   t-att-data-bs-toggle="wblog_nav_offcanvas and 'offcanvas' or 'dropdown'"
+                   t-att-data-bs-target="wblog_nav_offcanvas and '#o_wblog_offcanvas'"
+                   t-att-aria-expanded="not wblog_nav_offcanvas and 'false'"
+                   t-att-aria-controls="wblog_nav_offcanvas and 'o_wblog_offcanvas'">
+                    <i t-if="wblog_nav_offcanvas" class="fa fa-navicon" role="img"/>
+                    <t t-else="">
+                        <span t-if="blog" class="text-truncate" t-out="blog.name"/>
+                        <span t-else="" class="text-truncate">All</span>
+                    </t>
+                </a>
+                <div t-if="wblog_nav_offcanvas" id="o_wblog_offcanvas" class="offcanvas offcanvas-end d-lg-none">
+                    <div class="offcanvas-header">
+                        <h5 class="offcanvas-title my-0">Blogs</h5>
+                        <button type="button" class="btn-close" data-bs-dismiss="offcanvas" aria-label="Close"/>
+                    </div>
+                    <div class="offcanvas-body">
+                        <ul class="nav navbar-nav">
+                            <li class="nav-item">
+                                <a href="/blog" t-attf-class="nav-link #{(not blog) and 'active'}" t-att-title="website_blog_all_title">All</a>
+                            </li>
+                            <li t-foreach="blogs" t-as="nav_blog" class="nav-item">
+                                <a t-attf-href="/blog/#{slug(nav_blog)}"
+                                   t-attf-class="nav-link #{(blog and nav_blog.id == blog.id) and 'active' or ''}"
+                                   t-att-title="nav_blog.name">
+                                    <span t-field="nav_blog.name"/>
+                                </a>
+                            </li>
+                        </ul>
+                    </div>
+                </div>
+                <ul t-elif="len(blogs) > 1" class="dropdown-menu dropdown-menu-end">
+                    <li>
+                        <a href="/blog" t-attf-class="dropdown-item #{(not blog) and 'active'}" title="All Blogs">All</a>
+                    </li>
+                    <li t-foreach="blogs" t-as="nav_blog">
+                        <a t-attf-href="/blog/#{slug(nav_blog)}"
+                           t-attf-class="dropdown-item #{(blog and nav_blog.id == blog.id) and 'active' or ''}"
+                           t-att-title="nav_blog.name">
+                            <span t-field="nav_blog.name"/>
+                        </a>
+                    </li>
```

Comment:
In master maybe this part could be shared with the (existing before this PR) code? Seems like a copy paste
But maybe overkill for only 2 `<li/>`